jsx/Form.js

```diff
@@ -1739,6 +1739,147 @@ class LorisElement extends Component {
   }
 }
 
+/**
+ * Radio Component
+ * React wrapper for a <input type='radio'> element.
+ */
+class RadioElement extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.generateLayout = this.generateLayout.bind(this);
+  }
+
+  handleChange(e) {
+    this.props.onUserInput(this.props.name, e.target.value);
+  }
+
+  generateLayout() {
+    let layout = [];
+    let disabled = this.props.disabled ? 'disabled' : null;
+    let required = this.props.required ? 'required' : null;
+
+    const styleRow = {
+      display: 'flex',
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      width: '100%',
+    };
+    const styleColumn = {
+      display: 'flex',
+      flexDirection: 'column',
+      alignSelf: 'flex-start',
+      marginRight: '10px',
+    };
+    const styleContainer = {
+      paddingTop: '7px',
+      cursor: 'pointer',
+    };
+    const styleLabel = {
+      margin: 0,
+      color: '#064785',
+      cursor: 'pointer',
+    };
+    const styleInput = {
+      display: 'inline-block',
+      margin: '0 5px 0 5px',
+      cursor: 'pointer',
+    };
+
+    let content = [];
+    for (const key in this.props.items) {
+      if (this.props.items.hasOwnProperty(key)) {
+        content.push(
+          <div key={key}
+               style={styleColumn}>
+            <div style={styleContainer}>
+              <input
+                type='radio'
+                name={this.props.name}
+                value={this.props.items[key]}
+                id={key}
+                required={required}
+                disabled={disabled}
+                onChange={this.handleChange}
+                style={styleInput}
+              />
+              <label htmlFor={key}
+                     style={styleLabel}
+              >
+                {key}
+              </label>
+            </div>
+          </div>
+        );
+      }
+    }
+
+    layout.push(
+      <div key={this.props.name + '_key'}
+           style={styleRow}>
+        {content}
+      </div>
+    );
+
+    return layout;
+  }
+
+  render() {
+    let errorMessage = null;
+    let requiredHTML = null;
+    let elementClass = this.props.elementClass;
+    let required = this.props.required ? 'required' : null;
+
+    // Add required asterix
+    if (required) {
+      requiredHTML = <span className='text-danger'>*</span>;
+    }
+    // Add error message
+    if (this.props.errorMessage) {
+      errorMessage = <span>{this.props.errorMessage}</span>;
+      elementClass = this.props.elementClass + ' has-error';
+    }
+    // Generate layout
+    const layout = this.generateLayout();
+
+    return (
+      <div className={elementClass}>
+        <label className={'col-sm-3 control-label'}
+               htmlFor={this.props.id}>
```

Comment:
does this id prop need to be required if its being used here to connect the label to the input element? also, i think the id needs to be given to whichever element it's for

Reply:
@zaliqarosli I removed & tested it. Seems to be unnecessary. I think it was added when I thought the label needed to be associated with the input field but I think that's if doing something with css which I'm not.